### PR TITLE
remove URI components from metadata host service endpoint when generating .mlzconfig

### PR DESCRIPTION
**Pull request by glennmusa**
# Description

Today, users in air-gapped clouds have issues initializing terraform backends when using the full ARM Metadata Host Service Endpoint URI (e.g. "https://management.azure.com/") and instead require just the host name (e.g. "management.azure.com")

This change, when using the `deploy.sh` quickstart, removes the URI components from the result of `az cloud show --query endpoints.resourceManager` when the .mlzconfig file is generated so that just the hostname value is passed as a `metadata_host` argument during terraform backend initialization. See #241 for more information.

Before, a .mlzconfig generated from `deploy.sh` used to generate a key value pair like this:

```plaintext
mlz_metadatahost=https://management.azure.com/
```

Now, it should generate a value like this:

```plaintext
mlz_metadatahost=management.azure.com
```

...and that plays nicer across clouds and azurerm providers.

You can test this manually with the `generate_config_file.sh` script:

```bash
my_tenant_id=$(az account show --query tenantId --output tsv)
my_sub_id=$(az account show --query id --output tsv)
src/scripts/config/generate_config_file.sh -f mytest.mlzconfig -e public -z mytestenvname -l eastus -s $my_sub_id -t $my_tenant_id
```

## Issue reference

The issue this PR will close: #241 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles or validates correctly
* [x] BASH scripts have been validated using `shellcheck`
* [x] All tests pass (manual and automated)
* [x] The documentation is updated to cover any new or changed features
* [x] Markdown files have been linted using the recommended linter. (See `.vscode/extensions.json`.)
* [x] Relevant issues are linked to this PR

### CODE REVIEW

**Comment by glennmusa** [comment]
Waiting on confirmation of expected metadata host values before merging.

**Comment by Phydeauxman** [comment]
Looks great Glenn, thanks for the quick turn on this...and thanks for providing the quick test. I ran the test and got the expected outcome.

**Review by Phydeauxman — APPROVED** [review]
Looks great Glenn. Took me a little bit to follow your logic here but only because I have never used some of the techniques you are using.

**Pull request merged**
